changes/ocd-3964.sql

```diff
@@ -0,0 +1,687 @@
+   constraint form_item_fk foreign key (form_item_id)
+        references openchpl.form_item (id) match full
+        on update cascade on delete restrict
+);
+drop trigger if exists change_request_attestation_submission_response_audit on openchpl.change_request_attestation_submission_response;
+create trigger change_request_attestation_submission_response_audit after insert or update or delete on openchpl.change_request_attestation_submission_response for each row execute procedure audit.if_modified_func();
+drop trigger if exists change_request_attestation_submission_response_timestamp on openchpl.change_request_attestation_submission_response;
+create trigger change_request_attestation_submission_response_timestamp before update on openchpl.change_request_attestation_submission_response for each row execute procedure openchpl.update_last_modified_date_column();
+
+insert into openchpl.change_request_attestation_submission_response (change_request_attestation_submission_id, response_id, form_item_id, last_modified_user, deleted)
+select 
+	crar.change_request_attestation_submission_id,
+	(select ar.id 
+	from openchpl.allowed_response ar
+		inner join openchpl.attestation_valid_response avr
+		on ar.response = avr.response 
+	where avr.id = crar.attestation_valid_response_id),
+	-- This is a little hard coded
+	(select fi.id
+	from openchpl.form f
+	    inner join openchpl.form_item fi
+	    	on f.id = fi.form_id 
+	    inner join openchpl.question q
+	        on fi.question_id = q.id
+	    inner join openchpl.attestation a
+	        on q.question = a.description
+	    inner join openchpl.attestation_period ap
+	    	on ap.form_id = f.id
+	 where ap.id = 1
+	 and a.id = crar.attestation_id),
+	 crar.last_modified_user,
+	 crar.deleted 
+from openchpl.change_request_attestation_response crar
+where not exists (
+	select *
+	from openchpl.change_request_attestation_submission_response
+	where change_request_attestation_submission_id = crar.change_request_attestation_submission_id
+	and response_id = (
+		select ar.id 
+		from openchpl.allowed_response ar
+			inner join openchpl.attestation_valid_response avr
+			on ar.response = avr.response 
+		where avr.id = crar.attestation_valid_response_id)
+	and form_item_id = (
+	    select fi.id
+		from openchpl.form f
+	    	inner join openchpl.form_item fi
+	    		on f.id = fi.form_id 
+		    inner join openchpl.question q
+		        on fi.question_id = q.id
+		    inner join openchpl.attestation a
+		        on q.question = a.description
+		    inner join openchpl.attestation_period ap
+		    	on ap.form_id = f.id
+	 	where ap.id = 1
+	 	and a.id = crar.attestation_id)
+	 and deleted = crar.deleted);
+
+------------------------------------------------------------------------------------       
+------             DROP TEMP COLUMNS USED DURING DATA CONVERSION              ------
+------------------------------------------------------------------------------------
```

Comment:
Do you want to leave this section here commented out?

Reply:
I wanted to document somewhere what we can drop after this ticket is implemented.  Agree, that this might not be the best place.